lib/data/services/response_objects/error_response.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_template/data/services/response_error.dart';
+import 'package:flutter_template/nstack/nstack.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'error_response.freezed.dart';
+part 'error_response.g.dart';
+
+@freezed
+
+///The BE should provide one unique code for each error, in this case the
```

Comment:
I don't think we should require BE to return a specific error message. Our contract with any BE should only be with HTTP error codes, and convert those HTTP error codes to DomainError in the domain layer, those DomainError would then be forwarded to the Presentation layer.

I agree that this may not be the case for all the projects, as BE will differ, but I think that this would be an ideal error handling architecture.

Reply:
I get the point, but we should def support this format out of the box. It is part of the "API manifesto" to follow this convention, but I agree that we should gracefully fallback to response code.

Reply:
> > While many projects differ indeed, I have not worked with one that uses codes exclusively to handle errors.
> 
> For the projects in Monstarlab that I've worked on they always use specific HTTP status codes to communicate error per endpoint

I've had the complete opposite experience. Nevertheless handling endpoint errors solely based on status code is not scalable at all because you'll need different errors for different occurrences in the same endpoint (specifically in 400 errors).

one of the best documentation for errors that i've worked with so far is HoW https://heroesofworkapi-stg.azurewebsites.net/index.html if you want to take a look at it.

Reply:
Handling error in the front end only based on HTTP code won't cover every scenario. Typically, BE return error messages along with a Valid HttpCode. Other than 401 , there is a need of understanding of the error message from the response(custom code or error key) and act on that.

Reply:
> you'll need different errors for different occurrences in the same endpoint

This is only true in the Domain layer level!

Reply:
> This is only true in the Domain layer level!

I am not sure what do you mean, errors are being returned from the BE which will be handled in the data layer first

Reply:
> I am not sure what do you mean,

I think we already discussed this in the Monthly Flutter meeting: the idea was to split error objects in two types: ResponseErrors, those the ones that would map HTTP error/response codes, and DomainErrors, this would be domain-related errors returned/thrown by the use-cases/Domain/Presentation layer.

Reply:
> Are we discussing about splitting Error Objects here ?

It is stated in the OP!

Reply:
well I agree with the following 

> the idea was to split error objects in two types: ResponseErrors, those the ones that would map HTTP error/response codes, and DomainErrors, this would be domain-related errors returned/thrown by the use-cases/Domain/Presentation layer.

You can open a PR mapping the ResponseError to domain error. 🙇

Reply:
@atiqmlbd @johsoe @pedromassango so are you guys fine with merging this and later discussing the data to domain to presentation mapping?

Reply:
yes 👍